File: Code/JavaScript/client/src/messages/HVPSLiveDataMessage.js
export default class HVPSLiveDataMessage
{
 static toJSON(o){ 
    const r = {};
   r["tpe"]="ld";
    r["ov"]=o.outputVoltage;
    r["oc"]=o.outputCurrent;
    r["toe"]=o.totalOutputEnergy;
    r["fsv"]=o.firstStageVoltage;
    r["pc"]=o.peakPrimaryCurrent;
    return r;
 }
 static fromJSON(o){
    const r = {};
    r.outputVoltage=o["ov"];
    r.outputCurrent=o["oc"];
    r.totalOutputEnergy=o["toe"];
    r.firstStageVoltage=o["fsv"];
    r.peakPrimaryCurrent=o["pc"];
   return r;
 }
}

File: Code/C#/HVPSMessages/Messages/HVPSLiveDataMessage.cs
using Core.Messages.Messages;
using HVPSMessages.DataMemberNames.Messages;
using System.Runtime.Serialization;
using System.Text.Json.Serialization;

namespace HVPSMessages.Messages
{
    [DataContract]
    public class HVPSLiveDataMessage : TypedMessageBase
    {
        [JsonPropertyName(HVPSLiveDataMessageDataMemberNames.OutputVoltage)]
        [JsonInclude]
        [DataMember(Name = HVPSLiveDataMessageDataMemberNames.OutputVoltage)]
        public double OutputVoltage { get; protected set; }
        [JsonPropertyName(HVPSLiveDataMessageDataMemberNames.OutputCurrent)]
        [JsonInclude]
        [DataMember(Name = HVPSLiveDataMessageDataMemberNames.OutputCurrent)]
        public double OutputCurrent { get; protected set; }
        [JsonPropertyName(HVPSLiveDataMessageDataMemberNames.TotalOutputEnergy)]
        [JsonInclude]
        [DataMember(Name = HVPSLiveDataMessageDataMemberNames.TotalOutputEnergy)]
        public double TotalOutputEnergy { get; protected set; }
        [JsonPropertyName(HVPSLiveDataMessageDataMemberNames.FirstStageVoltage)]
        [JsonInclude]
        [DataMember(Name = HVPSLiveDataMessageDataMemberNames.FirstStageVoltage)]
        public double FirstStageVoltage { get; protected set; }
        [JsonPropertyName(HVPSLiveDataMessageDataMemberNames.PeakPrimaryCurrent)]
        [JsonInclude]
        [DataMember(Name = HVPSLiveDataMessageDataMemberNames.PeakPrimaryCurrent)]
        public double PeakPrimaryCurrent { get; protected set; }
        public HVPSLiveDataMessage()
            : base()
        {
            Type = MessageTypes.HVPSUILiveData;
        }
    }
}
